### ISSUE TITLE
[Proposal]: Power Platform + Azure

### ISSUE THREAD

**Issue by mattruma** [microsoft/WhatTheHack#385]
### Topic

The Power Platform + Azure WTH will provide a deep understanding of how to integrate the Power Platform with backend services in Azure, consisting of Cosmos, API Management and Azure Functions.

### Abstract/Learning Objectives

Participants will learn how to:

* Build a Power App canvas application
* Connect to backend APIs fronted by Azure API Management
* Upload/Download files to Azure Storage
* Use Cognitive Search to search a knowledge base
* Create a Power BI dashboard
* Create a Virtual Agent to schedule appointments

### Delivery Date

6/20/22

### Authors

```markdown
Matt Ruma (mattruma)
Preston Hale (ThePreston)
```

### Other

Rough outline of the proposed WTH can be found at https://gist.github.com/mattruma/93bc263f02e019de004da4479db48523.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mattruma** [comment]
# What The Hack - Equipment Maintenance with the Power Platform

Based loosely on https://docs.microsoft.com/en-us/powerapps/guidance/fusion-dev-ebook but with more of a manufacturing spin.

 ## Introduction

 Power Platform and Azure are better together.

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

When complete, you'll have implemented the following:

* Created a Canvas App that allows maintenance crews to:
  * Search inventory.
  * Search a knowledge base, with AI assistance.
  * Respond to appointments.
  * Order replacement parts.
* Created an automated process to order replacement parts that connects to the third-party ordering system.
* Created a virtual agent that allows plant managers to:
  * Search the knowledge base.
  * Schedule appointments.
* Created a dashboard to view that allows maintenance crew managers to:
  * View scheduled appointments for the current week.
  * View completed appointments for the current week.
  * View appointments on hold due to replacement parts on order.

## Learning Objectives

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

* Lorem ipsum dolor sit amet
* Duis aute irure dolor in reprehenderit
* Excepteur sint occaecat cupidatat non proiden

## Challenges

This is just a rough draft, may include more details then necessary, e.g. specific Azure resources.

* Challenge 0: **Install local tools, Azure prerequisites and Power Platform prerequisites**
  * Create a Power Platform Developer Plan.
  * Install the prerequisites tools and software.
  * Create the Azure resources required.
* Challenge 1: **Build low code prototypes**
  * Create Field inventory management prototype.
  * Create Field knowledge base prototype.
  * Create Field scheduling and notes prototype.
  * See https://docs.microsoft.com/en-us/powerapps/guidance/fusion-dev-ebook/03-building-low-code-prototype.
* Challenge 2: **Connect to Azure API**
  * Updating the app to use the connector: Field inventory management.
  * Updating the App to use the Connector: Field scheduling and notes.
* Challenge 3: **Display images parts**
* Challenge 4: **Upload a photograph to an appointment**
* Challenge 5: **Track and display appointment history for a customer**
* Challenge 6: **Connect Azure Cognitive Search service to the Field knowledgebase**
  * Create the Azure Cognitive Search service for the Field knowledgebase.
  * Creating the custom connector for the Azure Cognitive Search service.
  * Updating the app to use Azure Cognitive Search.
* Challenge 7: **Ordering parts**
  * Order parts from a vendor through a web hook.
  * Part fulfillment can take upwards to 30 seconds, the vendor allows for a web hook to be passed as a callback when the order has been fulfilled.
  * Solution could involve Logic Apps or Power Automate.
* Challenge 8: **Create virtual agent for plant managers to schedule service appointments**
* Challenge 9: **Create Power BI dashboard**
  * Display appointments scheduled for the current week.
  * Display appointments completed for the current week.
  * Display appointments being held up by parts on order.
* Challenge 10: **Deploy the application**
  * Share with other users.

Azure resources **included** in the initial setup: Azure App Service for the API, Azure APIM, Azure Storage and CosmosDB.

Azure resources that **could be used** in the solution: Logic Apps, Service Bus, Storage Queues, Event Grid and Azure Functions.

Power Platform resources that **should be used** in the solution: Canvas Apps, Connectors, Virtual Agents, Power BI and Power Automate.
